flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/config/ExecutionConfigOptions.java

```diff
@@ -435,6 +435,16 @@ public class ExecutionConfigOptions {
                             "Determines whether CAST will operate following the legacy behaviour "
                                     + "or the new one that introduces various fixes and improvements.");
 
+    @Documentation.TableOption(execMode = Documentation.ExecMode.BATCH_STREAMING)
+    public static final ConfigOption<MapKeyDedupPolicy> TABLE_EXEC_MAPKEY_DEDUP_POLICY =
+            key("table.exec.mapkey-dedup-policy")
```

Comment:
why should we have this and why this behavior is related to only one function?
There are other `MAP` related functionalities

Reply:
this is job level like spark, when it has merged, i will fix the map_from_array here https://issues.apache.org/jira/browse/FLINK-31682 
https://github.com/apache/spark/blob/7e2c6c7ab23f75a6ba83baaa0545482a43845ce8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/util/ArrayBasedMapBuilder.scala#L52

Reply:
besides `map_from_array` there are others, e.g. `map` constructor
different ways of casting e.g. 
```sql
select cast(map['1', '2', ' 1', '2'] as map<int,int>);
```

it looks like a major change and probably makes sense to discuss it first rather to make it silently (e.g. there is nothing about that in corresponding JIRA issue)

Reply:
yeap, all kinds of map should consider this in spark

the map_constructor here @snuyanzin 
https://github.com/apache/spark/blob/eeab2e701330f7bc24e9b09ce48925c2c3265aa8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/complexTypeCreator.scala#L184

Reply:
i will discuss in the mail list latter ,thanks @snuyanzin